### PR TITLE
Container infrastructure

### PR DESCRIPTION
**Pull request by pgrinaway**
To simplify debugging and provision of the code in this repo, this PR contains a Dockerfile along with some updates to conda dependencies. Right now, the user would still need to provide an OpenEye license (until #342 is resolved) to use most of the code. Marked WIP because I need to make sure this actually builds correctly (it did before, but without the required dependencies that I've now added).

### CODE REVIEW

**Comment by jchodera** [comment]
@pgrinaway : Installation of the openeye tools can now be done with:
```bash
conda install -c openeye openeye-toolkits
```
No need to add the `openeye` channel or install via `pip`. See more [here](https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html#installing-openeye-python-toolkits).

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rm -f $MINICONDA
+ENV PATH /miniconda/bin:$PATH
+
+# Add channels and build, then install
+RUN conda config --add channels omnia
+RUN conda update --yes -n base conda
+RUN conda config --add channels conda-forge
+RUN conda config --add channels omnia/label/dev
+RUN conda config --add channels openeye
+RUN pip install progressbar2
```
Comment (jchodera) [inline Dockerfile]: Why can't you use the conda-installable `progressbar2` from conda-forge?
https://anaconda.org/conda-forge/progressbar2

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+RUN MINICONDA="Miniconda3-latest-Linux-x86_64.sh" && \
+    wget --quiet https://repo.continuum.io/miniconda/$MINICONDA && \
+    bash $MINICONDA -b -p /miniconda && \
+    rm -f $MINICONDA
+ENV PATH /miniconda/bin:$PATH
+
+# Add channels and build, then install
+RUN conda config --add channels omnia
+RUN conda update --yes -n base conda
+RUN conda config --add channels conda-forge
```
Comment (jchodera) [inline Dockerfile]: After installing the `conda-forge` channel, you should use `conda update --yes --all`. 
See our recommendation for using `omnia` here:
http://www.omnia.md/install/

File: Dockerfile (hunk 1 above)
Comment (pgrinaway) [reply Dockerfile]: Changing now.

**Comment by jchodera** [comment]
@pgrinaway : If you're using the CUDA 9 image, you'll want to pin `openmm` to be `>=7.2.0`

File: Dockerfile (hunk 2 above)
Comment (pgrinaway) [reply Dockerfile]: Ok

**Comment by pgrinaway** [comment]
>: Installation of the openeye tools can now be done with:

Well, I wanted to add it as a dependency in the `meta.yaml` so that it is automatically installed. Presumably the channel needs to be added for that to happen?

**Comment by jchodera** [comment]
> Presumably the channel needs to be added for that to happen?

Ah, yes, sorry!

**Comment by pgrinaway** [comment]
>@pgrinaway : If you're using the CUDA 9 image, you'll want to pin openmm to be >=7.2.0

Ok, got it.

**Comment by pgrinaway** [comment]
Making sure dockerhub can build this, then removing the `-b ` option from the `git clone` and merging.

**Comment by pgrinaway** [comment]
Ok, the dockerhub build succeeded, merging.

**Pull request merged**